### DISCUSSION TITLE
Porting Mbed6 to Mbed-CE

### DISCUSSION THREAD

**Discussion by cakeslob** [mbed-ce/mbed-os#306] [General]
I am in the progress of porting over a multi target project from Mbed6 to Mbed CE, the original reasons were because I believe MbedCE will have better support for STM32H7/G0, and because  I am also hoping the  SPI DMA api will make it easier to add more targets to the project. Since the announcement from Mbed, I dont have a choice now. 

I apologize, but I am a noob when it comes to computer programming.  I have successfully compiled a hello world for the NucleoF446, and have moved on to porting the project

After playing wack a mole with compile errors, Ive encountered a problem I dont understand yet. 

The file structure of the mbed project I work on, is different than what is in Mbed-CE 

Here is the project I work on, 
https://github.com/cakeslob/Remora/tree/main/Firmware/FirmwareSource/Remora-OS6

Is there a reference you can point me to, as to how the files should be organized? 

I appreciate your help, and patience, as i fumble through the process.

**Comment by wdx04** [comment]
Mbed CE's build system is based on CMake, you'll need to transform custom targets and libraries into CMake library targets.
If you are familiar with CMake, you can take a look at my repository:
https://github.com/wdx04/mbed_ce_port

**Comment by JohnK1987** [comment]
It depends on how clean the structure of your project should be.
Oficial MbedCE [Custom target example is here](https://github.com/mbed-ce/mbed-ce-custom-targets). Inside you will found oficial MbedCE structure what is recommended if you are CMake beginner.
For libraries we have another organization https://github.com/mbed-ce-libraries-examples. You can look inside for a reference how to deal with your libraries in your project. For example - https://github.com/mbed-ce-libraries-examples/DS1820

Many things are descripted in [MbedCE wiki](https://github.com/mbed-ce/mbed-os/wiki). Not just via links on the main page but also via right side panel.

**Reply by JohnK1987** [reply]
@cakeslob any progress?

**Comment by cakeslob** [comment]
Yes, thanks guys. Im just reading/learning more about cmake file structure compared to what ever mbed6 is using for its file structure. I have some examples to work with now thought. but I will admit, mbed-ce is considerable harder than the prepackaged mbed. I need to read more about cmakelists , that is probably why it cant find my files.  If we measure progress by compile percent, I have made 5% progress. 

this is what it looks like so far, 
https://github.com/cakeslob/remora-mbed-ce
